### PR TITLE
ACMG modifiers - fix #2463

### LINKED ISSUE #2463: ACMG classification page
Hi , we at CMMS have a wish for this page.
We have read the attached publication where the PSV1 criteria can be downgraded to
PVS1-strong
PVS1-moderate
PVS1-supp

This downregulation affects the final classification (P, PL, VUS)

It would be great if this can be integrated in the page. 

[humu.23626.pdf](https://github.com/Clinical-Genomics/scout/files/6182432/humu.23626.pdf)

Is it possible?
See suggestion in the pp slide, if of any help 
[PVS1.pptx](https://github.com/Clinical-Genomics/scout/files/6182501/PVS1.pptx)

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

A tentative fix for adding ACMG term modifiers. There are a some things left (tests etc), but it is late and it is also working at the moment so pushing a draft.

This adds a modifier option to the classification terms so that e.g. this Very Strong, directly Pathogenic with another Strong:
Screenshot 2023-01-10 at 17 06 31
When modified to Moderate evidence gives a Likely Pathogenic verdict.
Screenshot 2023-01-10 at 17 06 24

See documentation update for more on what it is supposed to do.. 😄

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Should also fix #2700.

File: scout/build/acmg.py [hunk 1]
```diff
@@ -45,13 +45,9 @@ def build_evaluation(
...
     for info in criteria:
         criteria_obj = {}
-        # This allways has to exist
-        # We might want to check if the term is valid here...
-        criteria_obj["term"] = info["term"]
-        if "comment" in info:
-            criteria_obj["comment"] = info["comment"]
-        if "links" in info:
-            criteria_obj["links"] = info["links"]
+        for criterion_key in ["term", "comment", "links", "modifier"]:
```
Comment (northwestwitch) [inline scout/build/acmg.py:48]: 💯

File: scout/models/acmg_evaluation.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
     # All evaluations will have an author
     user_id=str,  # user email, required
     user_name=str,  # user name
-    criterias=list,  # List of dictionaries with criterias
```
Comment (northwestwitch) [inline scout/models/acmg_evaluation.py:23]: 🥇

File: scout/server/blueprints/variant/templates/variant/acmg.html [hunk 3]
```diff
@@ -75,12 +78,22 @@ <h4>Evidence of {{ category }}</h4>
                         <label class="form-check-label" for="checkbox-{{ criterion_code }}"></label>
                       </div>
                     </div>
-                    <div id="comment-{{ criterion_code }}" class="{{ 'collapse' if not (comment or link) }} mt-2">
+                    <div id="comment-{{ criterion_code }}" class="{{ 'collapse' if not (comment or link or modifier) }} mt-2">
+                      <div class="row">
+                        <select {{ 'disabled' if evaluation }} id="modifier-{{ criterion_code }}" name="modifier-{{ criterion_code }}" class="form-control form-select">
+                          <option value="" {% if not modifier %}selected{% endif %}>Strength modifier...</option>
+                          {% for level in "Strong", "Moderate", "Supporting" %}
+                            {% if(level != evidence) %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/acmg.html:86]: Does very strong need a 'strong' modifier?

image

Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/acmg.html:86]: And stand alone as well, for the benign ones:

image

**Review by northwestwitch — APPROVED** [review]
Brilliant solution to a problem raised in many labs using Scout! 🥇 

Looks good to me, code-wise and functionality!

File: docs/user-guide/acmg-criterias.md [hunk 4]
```diff
@@ -84,92 +84,104 @@ PM6 Assumed de novo, but without confirmation of paternity and maternity
...
+# Modifying criteria strength
+
+The ACMG criteria allow for modifying the weight of different criteria based on the
+strength of the evidence, or certain modifying information, with a lab manager's professional judgement.
+A common, well documented application is to downgrade `PVS1` variants (Tayoun et al, Human mutation 2019). Another would be the application of a modified segregation in the family based on different levels of evidence.
+Or the recognition that variants sometimes do not meet the strict frequency criteria of `PM2`, `BA1` or `BS1` but that a tendency could warrant a modified
+`PM2_Supporting` or `BS1_Supporting` (Harrison et al Curr Prot Hum Genet 2019, PMID: 31479589)[https://pubmed.ncbi.nlm.nih.gov/31479589/].
+
+We apply the modification nomenclature introduced by the [ClinGen variant interpretation working group in 2017](https://www.clinicalgenome.org/site/assets/files/3459/svi_criteria_nomenclature_recommendation_v1.pdf)
+Compare e.g. the [Baylor approved criteria](https://www.baylorgenetics.com/variant-classification/).
```
[the remark this answers is not in the document]
Comment (dnil) [reply docs/user-guide/acmg-criterias.md:186]: I liked a paper that came out recently that analyzed the schema from a Bayesian point of view and found that the ACMG heuristics mess for different levels gave a stable a posteriori probability for each level, give or take a little. It kind of works, and I guess having someone add the probabilities would just make it worse for the operator. Now, that does open for a semi-automatic solution.

File: scout/server/blueprints/variant/templates/variant/acmg.html (hunk 3 above)
Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/acmg.html:86]: A right, they actually do - it serves as a downgrade in this case. Like for instance if you have a frame shift but it’s rather late in the gene though not in the very final part. Then perhaps Strong is better than Very Strong.

Conversely,  the scheme could use Stand Alone or Very Strong modifiers for other criteria but I’ve never seen it in any paper or the need for it in practice.

**Pull request merged**